app/shop.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass
+from app.customer import Customer
+from app.car import Car
+import math
+# from datetime import datetime
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list
+    products: dict
+
+
+def calculate_trip_price(
+        fuel_price: float,
+        client: Customer,
+        shop: Shop,
+        car: Car) -> float:
+    distance = math.sqrt(
+        ((client.location[0] - shop.location[0]) ** 2)
+        + ((client.location[1] - shop.location[1]) ** 2))
+    trip_price = fuel_price * distance * (car.fuel_consumption / 100) * 2
+    for product, price in shop.products.items():
+        if product in client.products_cart:
+            trip_price += (price * client.products_cart.get(product))
+    trip_price = round(trip_price, 2)
+    print(f"{client.name}'s trip to the {shop.name} costs {trip_price}")
+    return trip_price
+
+
+def check_best_price(
+        trip_prices: dict,
+        client: Customer,
+        shops: dict) -> None:
+    best_choice = min(trip_prices.values())

```

Comment:
You can return key here, for example 

`best_choice = min(trip_prices, key=trip_prices.get)`
Then you won't need to iterate through trip_prices again